source/common/common/logger.h

```diff
@@ -116,7 +116,7 @@ template <Id id> class Loggable {
   /**
    * @return spdlog::logger& the static log instance to use for class local logging.
    */
-  static spdlog::logger& log() {
+  static spdlog::logger& logger() {
```

Comment:
Haven't fully looked into this yet, but naively, why can't we use appropriate namespace qualification and/or if we do this rename, make it something we'll never conflict with, e.g. a `_` prefix or something?

Reply:
Agreed, I actually had meant to rename this something like: `__log__do_not_use_read_comment()` or something like that. Then tell people to use ENVOY_LOG.

Reply:
@mattklein123 @htuch I'm fine with whatever name works with `Http::AccessLog::Instance`. I'll change this to `__log_do_not_use_read_comment`.